### ISSUE TITLE
Does not overwrite when run from NeoVim (:YcmGenerateConfig)

### ISSUE THREAD

**Issue by waldnercharles** [rdnetto/YCM-Generator#48]
`:YcmGenerateConfig` will prompt to overwrite, but will return 1 immediately when in NeoVim.

Is this expected behavior?

**Comment by rdnetto** [comment]
No, interactive behaviour is expected when run from Vim. Looks like this is due to a difference in how Neovim spawns processes - run `:! cat` terminates immediately as well. As explained in neovim/neovim#1477, this is because child processes do not get access to tty (since that would break the msgpack-rpc UI).

It's possible to workaround this using `:terminal` and the new job control API (I actually implemented this as an experiment), but on reflection it makes more sense to simply change the command to not overwrite unless it's suffixed with `!` (for consistency with `:w!`, etc.)

**Comment by rdnetto** [comment]
I've pushed a fix for this to `develop` - can you verify it's working?

**Comment by Yanhao** [comment]
I got the same issue in with vim(not neovim)

**Comment by rdnetto** [comment]
@YanhaoMo Please open a separate bug for that - Vim and Neovim handle child processes differently.

**Comment by a1lu** [comment]
develop branch is not working for me
